riscv/insns/amoadd_b.h

```diff
@@ -0,0 +1,3 @@
+require_extension('A');
+require_extension(EXT_ZABHA);
+WRITE_RD((sreg_t)(int8_t)(MMU.amo<uint8_t>(RS1, [&](uint8_t lhs) { return lhs + RS2; })));
```

Comment:
Any reason we can't simplify this to the following:

    WRITE_RD(sreg_t(MMU.amo<int8_t>(RS1, [&](int8_t lhs) { return lhs + RS2; })));

If not, let's do the same for everything in Zabha.

Reply:
The type cast to `int8_t` was to do the sign extension and is required. I have updated it to:
         `WRITE_RD(sreg_t((int8_t)MMU.amo<int8_t>(RS1, [&](int8_t lhs) { return lhs + RS2; })));`

Reply:
Understood that sign extension is the reason, but you can delete the int8_t cast, because after this change, the function’s return type is already int8_t.

Reply:
@aswaterman - I missed that part. I have updated to remove the explicit cast.